### PR TITLE
removing unused code

### LINKED ISSUE #2977: BUG: self.history in GLM doesn't seem like it does anything...
Hello,

The `history` attribute in GLM doesn't seem like it really does anything. It seems like what's happening is that the history is created by each of the district `fit` methods. 

See this line of code:
https://github.com/statsmodels/statsmodels/blob/master/statsmodels/genmod/generalized_linear_model.py#L291

You can uncomment it all and GLMs seem to work just as before. A few tests fail when they try to reference that attribute, however, so there's some hunting that will be necessary to clean this up.

### PR DESCRIPTION
**Pull request by thequackdaddy**
closes #2977 

This code doesn't do anything... I think it can be safely removed.

### CODE REVIEW

File: statsmodels/genmod/generalized_linear_model.py [hunk 1]
```diff
@@ -1150,8 +1145,6 @@ def fit_constrained(self, constraints, start_params=None, **fit_kwds):
         res._results.constraints = lc
         res._results.k_constr = k_constr
         res._results.results_constrained = res_constr
-        # TODO: the next is not the best. history should bin in results
-        res._results.model.history = res_constr.model.history
```
Comment (bashtage) [inline statsmodels/genmod/generalized_linear_model.py]: Is `res_constr.model.history` always empty?

**Comment by jbrockmendel** [comment]
LGTM

**Comment by josef-pkt** [comment]
to my inline comment: 
reviewing mle_retvals and similar is a different issue and doesn't need to be here. #2401

we just need `fit_history`

File: statsmodels/genmod/generalized_linear_model.py (hunk 1 above)
Comment (jbrockmendel) [reply statsmodels/genmod/generalized_linear_model.py]: > that means [...] should work as a replacement

I'm not entirely clear, is this actionable for the OP?

The only strong opinion I have is that the Model instance should not be changed by the call to `fit`.

**Comment by jbrockmendel** [comment]
@thequackdaddy needs rebase

**Pull request closed without merging**